src/Servers/HttpSys/src/RequestProcessing/Request.cs

```diff
@@ -323,6 +326,30 @@ private void GetTlsHandshakeResults()
             KeyExchangeStrength = (int)handshake.KeyExchangeStrength;
         }
 
+        public X509Certificate2 ClientCertificate
+        {
+            get
+            {
+                if (_clientCert == null && SslStatus == SslStatus.ClientCert)
+                {
+                    try
+                    {
+                        _clientCert = _nativeRequestContext.GetClientCertificate();
+                    }
+                    catch (CryptographicException ce)
+                    {
+                        RequestContext.Logger.LogDebug(ce, "An error occurred reading the client certificate.");
```

Comment:
nit: separate method to call LogDebug s.t. you can remove the duplication. Also is Debug the right verbosity?

Reply:
It's one line. A separate method would be more.

Debug is what we use for client errors the server doesn't control.